apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +58,107 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteSlots = (paletteSlots: IPalette): IPaletteSlots => {
+    let trimmedPaletteSlots: IPaletteSlots = {};
+    for (let palette in paletteSlots) {
+      if (
+        palette.startsWith('theme') ||
+        palette.startsWith('neutral') ||
+        palette === 'black' ||
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        trimmedPaletteSlots[palette] = (paletteSlots as any)[palette];
+      }
+    }
+    return trimmedPaletteSlots;
+  };
+
+  const isASemanticColor = (semanticString: string): boolean => {
+    return (
+      semanticString.startsWith('body') ||
+      semanticString.startsWith('primary') ||
+      semanticString.startsWith('disabled') ||
+      semanticString.startsWith('focus') ||
+      semanticString.startsWith('variant') ||
+      semanticString.startsWith('default') ||
+      semanticString.includes('action') ||
+      semanticString.includes('link') ||
+      semanticString.startsWith('accent') ||
+      semanticString.startsWith('list') ||
+      semanticString.startsWith('button') ||
+      semanticString.startsWith('menu') ||
+      semanticString.startsWith('input') ||
+      semanticString.startsWith('small')
+    );
+  };
+
+  const trimSemanticSlotsOrNames = <T extends unknown>(semanticSlots: T) => {
+    if (semanticSlots instanceof Array) {
+      let trimmedSemanticSlotNames = [];
+      for (let i = 0; i < semanticSlots.length; i++) {
+        let slotName = semanticSlots[i];
+        if (isASemanticColor(slotName)) {
+          trimmedSemanticSlotNames.push(slotName);
+        }
+      }
+      return trimmedSemanticSlotNames;
+    } else {
+      let trimmedSemanticSlots: ISemanticSlots = {};
+      for (let semanticColor in semanticSlots) {
+        if (isASemanticColor(semanticColor)) {
+          trimmedSemanticSlots[semanticColor] = (semanticSlots as any)[semanticColor];
+        }
+      }
+      return trimmedSemanticSlots;
+    }
+  };
+
+  const fillVariantSlotsList = (variantType: VariantThemeType): JSX.Element[] => {
     let currThemeVariant: ITheme;
+    let noneVariant = getVariant(props.theme, VariantThemeType.None);
     if (props.theme) {
```

Comment:
I think this would still work even with an undefined theme?

Reply:
yes moving these 2 lines above inside the if - thanks for catching this!